lldb/source/Plugins/LanguageRuntime/Swift/SwiftLanguageRuntimeNames.cpp

```diff
@@ -589,18 +590,50 @@ static lldb::ThreadPlanSP GetStepThroughTrampolinePlan(Thread &thread,
         thread, sym_addr_range, sc, function_name.c_str(), eOnlyDuringStepping,
         eLazyBoolNo, eLazyBoolNo);
   }
+  case ThunkAction::StepIntoAllocatingInit: {
+    LLDB_LOGF(log, "Stepping into allocating init: \"%s\"", symbol_name);
+    swift::Demangle::Context ctx;
+    NodePointer demangled_node =
+        SwiftLanguageRuntime::DemangleSymbolAsNode(symbol_name, ctx);
+
+    using Kind = Node::Kind;
+    NodePointer class_node = childAtPath(
```

Comment:
Do you need to check that demangled_node is non-null before calling childAtPath?

Reply:
Oh, the function `childAtPath` is safe, it checks for nullptr. (If it didn't, we would have changed it to take a reference instead)

Reply:
Okay.